### PR TITLE
Add script to report numpy env

### PR DESCRIPTION
**Pull request by levbishop**
For tracking down subtle environment-related issues, we can run this script to  get some details

### CODE REVIEW

**Review by jakelishman — APPROVED** [review]
I'm happy with this and think it'll be super helpful for debugging, but I pushed up the changes to the CI jobs to make this actually run (at Lev's request), so somebody else should approve before this merges.

**Comment by levbishop** [comment]
FWIW I'm completely happy with Jake's updates. If you want to see the results of this change, check out:
https://dev.azure.com/qiskit-ci/qiskit/_build/results?buildId=54644&view=logs&j=42553bb7-c640-5e19-486b-534ade7d1f3b&t=ec81b5c9-a7aa-58a8-95bd-450209a80da7&l=12

**Review by mtreinish — APPROVED** [review]

**Pull request merged**
